Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -851,9 +931,294 @@ struct HFTrackIndexSkimsCreator {
+        MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "Fitter result = " << nCand2 << " proton = " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg);
+        MY_DEBUG_MSG(isLc, LOG(INFO) << "Fitter result for true Lc = " << nCand2);
+        if (nCand2 == 0) {
+          continue;
+        }
+        std::array<float, 3> pVecCandCasc = {0., 0., 0.};
+        fitter.propagateTracksToVertex();        // propagate the bach and V0 to the Lc vertex
+        fitter.getTrack(0).getPxPyPzGlo(pVecV0); // take the momentum at the Lc vertex
+        fitter.getTrack(1).getPxPyPzGlo(pVecBach);
+
+        pVecCandCasc = array{pVecBach[0] + pVecV0[0],
+                             pVecBach[1] + pVecV0[1],
+                             pVecBach[2] + pVecV0[2]};
+
+        // cascade candidate pT cut
+        if (RecoDecay::Pt(pVecCandCasc) < d_cutCascPtCandMin) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg << " rejected due to pt cut: " << RecoDecay::Pt(pVecCandCasc) << " (cut " << d_cutCascPtCandMin << ")");
+          continue;
+        }
+
+        // invariant mass
+        // calculate invariant masses
+        auto arrMom = array{pVecBach, pVecV0};
+        mass2K0sP = RecoDecay::M(arrMom, array{massP, massK0s});
+        // invariant-mass cut
+        if ((d_cutCascInvMassLc >= 0.) && (std::abs(mass2K0sP - massLc) > d_cutCascInvMassLc)) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc, LOG(INFO) << "True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg << " rejected due to invMass cut: " << mass2K0sP << ", mass Lc " << massLc << " (cut " << d_cutCascInvMassLc << ")");
+          continue;
+        }
+
+        std::array<float, 3> posCasc = {0., 0., 0.};
+        const auto& cascVtx = fitter.getPCACandidate();
+        for (int i = 0; i < 3; i++) {
+          posCasc[i] = cascVtx[i];
+        }
+
+        // fill table row
+        rowTrackIndexCasc(bach.globalIndex(),
+                          v0.globalIndex(),
+                          1); // 1 should be the value for the Lc
+        // fill histograms
+        if (b_doValPlots) {
+          MY_DEBUG_MSG(isK0SfromLc && isProtonFromLc && isLc, LOG(INFO) << "KEPT! True Lc from proton " << protonLabel << " and K0S pos " << labelPos << " and neg " << labelNeg);
+          registry.get<TH1>(HIST("hvtx2_x"))->Fill(posCasc[0]);
+          registry.get<TH1>(HIST("hvtx2_y"))->Fill(posCasc[1]);
+          registry.get<TH1>(HIST("hvtx2_z"))->Fill(posCasc[2]);
+          registry.get<TH1>(HIST("hmass2"))->Fill(mass2K0sP);
+        }
+
+      } // loop over V0s
+
+    } // loop over tracks
+  }   // process
+};
+
+struct HFTrackIndexTestMCCasc {
+  void process(aod::BigTracksMC const& tracks,
+               aod::V0Datas const& V0s,
+               aod::McParticles const& particlesMC)
```

Comment:
What is this device supposed to do? It seems to just loop and match without any output.

Reply:
It was used to test the MC matching for the K0S daughters. A sort of debug task. I will remove it.